### ISSUE TITLE
Helm3: No 'init', doesn't use existing ~/.helm

### ISSUE THREAD

**Issue by joelbernstein** [helm/helm#6359]
`./helm version
version.BuildInfo{Version:"v3.0.0-beta.1", GitCommit:"f76b5f21adb53a85de8925f4a9d4f9bd99f185b5", GitTreeState:"clean", GoVersion:"go1.12.9"}
`

I've upgraded from helm v2 by downloading the beta v3 client.
I've deleted the tiller account with the v2 client.
No helm commands using v3 seem to work -- it knows nothing about repositories. 
If I supply them on command line, it still doesn't DTRT:
```
➜  helm-darwin-amd64 ./helm --repository-cache ~/.helm/repository/cache --repository-config ~/.helm/repository/repositories.yaml repo update
Error: couldn't load repositories file (): open : no such file or directory
```

But it clearly can read the config therein:
```
➜  helm-darwin-amd64 ./helm --repository-cache ~/.helm/repository/cache --repository-config ~/.helm/repository/repositories.yaml repo list
NAME  	URL
stable	https://kubernetes-charts.storage.googleapis.com
local 	http://127.0.0.1:8879/charts
gitlab	https://charts.gitlab.io/
```

But installing any chart fails:
```
➜  helm-darwin-amd64 ./helm --repository-cache ~/.helm/repository/cache --repository-config ~/.helm/repository/repositories.yaml install stable/kong --set ingressController.enabled=true -g
Error: failed to download "stable/kong" (hint: running `helm repo update` may help)
[1]    66253 exit 1     ./helm --repository-cache ~/.helm/repository/cache --repository-config
```

Why not add a virtual subcommand `helm init` like in v2, which either reconfigures helm based on v2 config, or explains properly how to modify the existing config? This is really not obvious. The docs on the v3.helm.sh/docs all refer to v2 and talk about `helm init` and so on. If the docs are this unready then it's not in Beta, surely? This is just silly.

Can you please point out the docs that explain how to upgrade to v3 without losing all functionality? Surely they exist?

**Comment by hickeyma** [comment]
@joelbernstein It looks like you might be using beta.1. There was some issues with this release and are fixed in [beta.2](https://github.com/helm/helm/releases/tag/v3.0.0-beta.2).  Can you try using that instead and see if it works as expected? FYI, default repos like `stable` are no longer set-up OOTB.

The upgrade/migration path is still WIP: https://github.com/helm/helm/issues/5892

**Comment by joelbernstein** [comment]
@hickeyma thanks for getting back to me. You're right, it looks like I'm using beta.1 which I guess was linked off some docs. I'll try with beta.2 and comment back.
Re upgrade/migration, i think that's to do with differences in user-supplied chart syntax? Whereas I'm just trying to use helmv3 to deploy some packaged code (e.g. Kong proxy) to my kubernetes cluster without needing RBAC and a Tiller namespace and so on. Is this a reasonable expectation on my part?

**Comment by joelbernstein** [comment]
@hickeyma that worked! Thanks
```
➜  helm-darwin-amd64 ./helm repo update
Error: no repositories found. You must add one before updating
[1]    66403 exit 1     ./helm repo update
➜  helm-darwin-amd64 ./helm --repository-cache ~/.helm/repository/cache --repository-config ~/.helm/repository/repositories.yaml repo update
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
	Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: connect: connection refused
...Successfully got an update from the "gitlab" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈```

**Comment by hickeyma** [comment]
@joelbernstein The migration is for using any configuration and releases from Helm v2. If you just want to use the cluster you can use Helm v3 side-by-side with Helm v2 without removing Tiller etc. More details about use cases: https://github.com/helm/helm/pull/5582

**Comment by joelbernstein** [comment]
@hickeyma by "releases", do you mean "any charts created and published with helm v2"?

I've just - experimentally - used helmv3b2 to deploy `stable/kong` and it seems to have worked fine.

**Comment by hickeyma** [comment]
> by "releases", do you mean "any charts created and published with helm v2"?

Yes

You should be good to go then.

**Comment by hickeyma** [comment]
Closing @joelbernstein. If you find any other issues, just raise an issue for that.

**Comment by tammersaleh** [comment]
I'm also using the helm v3.0.0-beta2 client, but from a completely fresh state (no `~/.helm` directory).  How do I add the correct repositories without the `helm init` command?

**Comment by tammersaleh** [comment]
Found this in another issue:  

``` bash
helm repo add stable https://kubernetes-charts.storage.googleapis.com
```

**Comment by joelbernstein** [comment]
> Closing @joelbernstein. If you find any other issues, just raise an issue for that.

Thanks, hope this comment doesn't reopen the issue (!) just wanted to say that I really approve of the design changes in helmv3 and I'm sorry if this issue wasted your time.

**Comment by joelbernstein** [comment]
FWIW one of the docs linked to v3beta1 but I can't recall which one. If I find it I'll send a PR to fix it.

**Comment by hickeyma** [comment]
@tammersaleh Docs have good info also, like: https://helm.sh/docs/using_helm/#helm-repo-working-with-repositories

**Comment by rimusz** [comment]
@tammersaleh check out my blog post https://rimusz.net/helm_v3_beta on how to use helm v3 and changes it introduced

**Comment by tammersaleh** [comment]
Thanks @hickeyma - The docs do list which url each repo points to, but doesn't quite explain what @rimusz' post does: that `init` was removed, and exactly how to add `stable` on a fresh helm3 installation.

**Comment by bacongobbler** [comment]
The reason why the documentation doesn't explain how to add back the `stable` repository is because the charts team are planning to deprecate that repository. The cloud bucket hosting that chart repository will eventually be shut down and will no longer host charts.

Even though Helm is designed to work with numerous distributed repositories, the stable repository has given the impression or feel of a "central" repository. The central repository has become more than a handful to manage and scale so it isn't going to work for the next phase of growth in the community.

This is where [Helm Hub](https://hub.helm.sh/) comes in. We want to encourage people to host their own chart repositories and share them in a more discoverable place. The charts team is working on that migration plan, and we will have more information to share once that is ready.

RE: the docs not explaining this, thanks for pointing that out. It's on our roadmap to update the documentation prior to the 3.0 final release. We've been focusing on bug squashing, but once [Helm Summit](https://events.linuxfoundation.org/events/helm-summit-2019/) is finished, we should have breathing room to update the documentation 

Anyways, this is starting to get off topic from OP's discussion. Feel free to ask these questions on Slack if you have further questions about that decision. Thanks!

**Comment by tammersaleh** [comment]
Thanks @bacongobbler, that explanation is very helpful!  Sorry if I came off as unhappy with the docs situation - I'm using a beta pre-release, so I completely understand the lack of docs.

**Comment by rimusz** [comment]
we still should properly document the stable remove removal and how to use it in the mean time till it gets shut down, as right now not documenting its removal properly I see it makes a big confusion that helm v3 beta is not working

**Comment by bacongobbler** [comment]
> RE: the docs not explaining this, thanks for pointing that out. It's on our roadmap to update the documentation prior to the 3.0 final release. We've been focusing on bug squashing, but once [Helm Summit](https://events.linuxfoundation.org/events/helm-summit-2019/) is finished, we should have breathing room to update the documentation

**Comment by adampl** [comment]
`helm repo add stable https://kubernetes-charts.storage.googleapis.com/`

**Comment by defyjoy** [comment]
> `helm repo add stable https://kubernetes-charts.storage.googleapis.com/`

Sorry for being noob . But isn't this supposed to be included by default OR isn't it supposed to be documented ? Is it my fault that I did not find this anywhere in documentation OR is it not there at all ?

**Comment by bacongobbler** [comment]
https://helm.sh/docs/intro/quickstart/#initialize-a-helm-chart-repository
